### ISSUE TITLE
Match legacy wakatime-cli behavior around logging failed network connections

### ISSUE THREAD

**Issue by alanhamlett** [wakatime/wakatime-cli#495]
Now that #385 is fixed, we're getting a lot of diagnostics logs from connection timeouts and other normal expected errors. For these "wontfix" errors, we should prevent sending diagnostics. The best way to do this is match the behavior of legacy wakatime-cli and only log errors when:

* offline db disabled
* saving the heartbeat to the offline db failed
* debug mode enabled

For reference, the legacy Python logic around logging failed network requests:
https://github.com/wakatime/wakatime/blob/master/wakatime/api.py#L123

**Comment by alanhamlett** [comment]
Example error messages:

```
{"caller":"/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/run.go:189","func":"runCmd","level":"error","message":"failed to run command: failed to send heartbeat(s): failed to send heartbeats via api client: failed making request to \"https://api.wakatime.com/api/v1/users/current/heartbeats.bulk\": Post \"https://api.wakatime.com/api/v1/users/current/heartbeats.bulk\": dial tcp 68.183.248.166:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.","now":"2021-07-06T14:37:51+08:00","version":"v1.18.9"}
```

```
{"caller":"/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/run.go:189","func":"runCmd","level":"error","message":"failed to run command: failed to send heartbeat(s): failed to send heartbeats via api client: failed making request to \"https://api.wakatime.com/api/v1/users/current/heartbeats.bulk\": Post \"https://api.wakatime.com/api/v1/users/current/heartbeats.bulk\": net/http: TLS handshake timeout","now":"2021-07-06T14:37:47+08:00","version":"v1.18.9"}
```

```
{"caller":"/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/run.go:189","func":"runCmd","level":"error","message":"failed to run command: failed to send heartbeat(s): failed to send heartbeats via api client: failed making request to \"https://api.wakatime.com/api/v1/users/current/heartbeats.bulk\": Post \"https://api.wakatime.com/api/v1/users/current/heartbeats.bulk\": dial tcp 68.183.248.166:443: i/o timeout (Client.Timeout exceeded while awaiting headers)","now":"2021-07-06T13:37:41+07:00","version":"v1.18.9"}
```

```
{"caller":"/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/run.go:189","func":"runCmd","level":"error","message":"failed to run command: failed to fetch today summary: failed fetching summaries from api: failed to make request to \"https://api.wakatime.com/api/v1/users/current/summaries\": Get \"https://api.wakatime.com/api/v1/users/current/summaries?end=2021-07-06&start=2021-07-06\": dial tcp 68.183.248.166:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.","now":"2021-07-06T14:37:44+08:00","version":"v1.18.9"}
```

**Comment by alanhamlett** [comment]
Receiving the above diagnostics is a side-effect of the bug, where we shouldn't be logging network problems as errors unless debug mode is enabled or offline mode disabled.

**Comment by gandarez** [comment]
Closed via #502
